android/greengrass-nucleus-android/src/main/java/com/aws/greengrass/android/provision/BaseProvisionManager.java

```diff
@@ -0,0 +1,452 @@
+                                path.resolve(file).toString());
+                    }
+                }
+            }
+        } catch (IOException e) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Gets BaseProvisionManager.
+     *
+     * @param filesDir path to files/ in application's directory
+     */
+    public static BaseProvisionManager getInstance(File filesDir) {
+        return provisionManagerMap.computeIfAbsent(filesDir, c -> new BaseProvisionManager(filesDir));
+    }
+
+    /**
+     * Creates BaseProvisionManager.
+     *
+     * @param filesDir path to files/ in application's directory
+     */
+    private BaseProvisionManager(File filesDir) {
+        logger = LogHelper.getLogger(filesDir, getClass());
+        rootPath = WorkspaceManager.getInstance(filesDir).getRootPath().toString();
+    }
+
+    /**
+     * Checking is the Nucleus already provisioned.
+     *
+     * @return result true if Nucleus is already provisioned
+     */
+    @Override
+    public boolean isProvisioned() {
+        boolean provisioned = false;
+        // Check config.yaml
+        try (InputStream inputStream = Files.newInputStream(
+                Paths.get(WorkspaceManager.getConfigPath().toString(), CONFIG_YAML_FILE))) {
+            Yaml yaml = new Yaml();
+            HashMap yamlMap = yaml.load(inputStream);
+            // Access HashMaps and ArrayList by key(s)
+            HashMap system = (HashMap) yamlMap.get(DeviceConfiguration.SYSTEM_NAMESPACE_KEY);
+            String certPath = (String) system.get(DeviceConfiguration.DEVICE_PARAM_CERTIFICATE_FILE_PATH);
+            String privKeyPath = (String) system.get(DeviceConfiguration.DEVICE_PARAM_PRIVATE_KEY_PATH);
+            String rootCaPath = (String) system.get(DeviceConfiguration.DEVICE_PARAM_ROOT_CA_PATH);
+
+            if (!Utils.isEmpty(certPath)
+                    && !Utils.isEmpty(privKeyPath)
+                    && !Utils.isEmpty(rootCaPath)) {
+                provisioned = true;
+            }
+        } catch (FileNotFoundException e) {
+            logger.atWarn().log("Couldn't find {} file.", CONFIG_YAML_FILE);
+        } catch (NoSuchFileException e) {
+            logger.atWarn().log("File {} doesn't exist.", CONFIG_YAML_FILE);
+        } catch (Exception e) {
+            logger.atWarn().log("File {} doesn't have provisioning configuration",
+                    CONFIG_YAML_FILE);
+        }
+        return provisioned;
+    }
+
+    /**
+     * Reset Nucleus config files.
+     */
+    @Override
+    public void clearNucleusConfig() {
+        deleteConfigRecursive(new File(WorkspaceManager.getConfigPath().toString()));
+    }
+
+    /**
+     * Reset Nucleus provisioning.
+     */
+    @Override
+    public void clearProvision() {
+        new File(String.format("%s/%s", WorkspaceManager.getConfigPath().toString(), CONFIG_YAML_FILE)).delete();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of deleting a file on this line should be checked and failure should be logged. java.io.File.delete() will simply return false if it fails to delete the file, for example, when the file does not exist. It will not throw unless there is a security exception.